src/Notify/Client/NotificationClient.cs

```diff
@@ -4,190 +4,204 @@
+                                           from value in nvc.GetValues(key)
+                                           select string.Format("{0}={1}",
+                                           WebUtility.UrlEncode(key),
+                                           WebUtility.UrlEncode(value));
+            return "?" + string.Join("&", segments);
+        }
 
-            if (!String.IsNullOrWhiteSpace(templateType))
+        public NotificationList GetNotifications(string templateType = "", string status = "", string reference = "",
+            string olderThanId = "")
+        {
+            var query = new NameValueCollection();
+            if (!string.IsNullOrWhiteSpace(templateType))
             {
                 query.Add("template_type", templateType);
             }
 
-            if (!String.IsNullOrWhiteSpace(status))
+            if (!string.IsNullOrWhiteSpace(status))
             {
                 query.Add("status", status);
             }
 
-            if (!String.IsNullOrWhiteSpace(reference))
+            if (!string.IsNullOrWhiteSpace(reference))
             {
                 query.Add("reference", reference);
             }
 
-            if (!String.IsNullOrWhiteSpace(olderThanId))
+            if (!string.IsNullOrWhiteSpace(olderThanId))
             {
                 query.Add("older_than", olderThanId);
             }
 
-            String finalUrl = "v2/notifications?" + query.ToString();
+            var finalUrl = GET_ALL_NOTIFICATIONS_URL + ToQueryString(query);
+            var response = GET(finalUrl);
 
-            String response = this.GET(finalUrl);
-
-            NotificationList notifications = JsonConvert.DeserializeObject<NotificationList>(response);
+            var notifications = JsonConvert.DeserializeObject<NotificationList>(response);
             return notifications;
         }
 
-        public TemplateList GetAllTemplates(String templateType = "")
+        public TemplateList GetAllTemplates(string templateType = "")
         {
-            String finalUrl = string.Format(
+            var finalUrl = string.Format(
                 "{0}{1}",
                 GET_ALL_TEMPLATES_URL,
-                templateType == "" ? "" : TYPE_PARAM + templateType
+                templateType == string.Empty ? string.Empty : TYPE_PARAM + templateType
             );
 
-            String response = this.GET(finalUrl);
+            var response = GET(finalUrl);
 
-            TemplateList templateList = JsonConvert.DeserializeObject<TemplateList>(response);
+            var templateList = JsonConvert.DeserializeObject<TemplateList>(response);
 
             return templateList;
         }
 
-        public ReceivedTextListResponse GetReceivedTexts(String olderThanId = "")
+        public ReceivedTextListResponse GetReceivedTexts(string olderThanId = "")
         {
-            String finalUrl = String.Format(
+            var finalUrl = string.Format(
```

Comment:
for the string.format it would usually use String.format 

http://peterkellner.net/2007/12/29/incsharpwhentousestringversesstring/